src/multiplex/server.rs

```diff
@@ -185,6 +185,13 @@ where
     type Output = Result<(), Error<T, S>>;
 
     fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+        #[cfg(feature = "tracing")]
+        let span = tracing::trace_span!("poll");
+        #[cfg(feature = "tracing")]
+        let _guard = span.enter();
+        #[cfg(feature = "tracing")]
+        tracing::trace!("poll");
```

Comment:
Nit/TIOLI: You may find it easier to have crate-local macros that are conditionally either re-exports of the`tracing` macros or no-ops (and a similar span type that's either a re-export of `tracing::Span` or a no-op empty struct). That way, you don't need to put conditional compilation attributes on every tracing thing in the main code.

Reply:
tbh I'm considering just making the `tracing` feature non-optional. It's a little annoying to replicate the macros because I would also want it to be able to parse the same kinds of fields and such as the "real" one.